### DISCUSSION TITLE
Is there a stable hostfxr ABI?

### DISCUSSION THREAD

**Discussion by mvtec-bergdolll** [dotnet/runtime#86100] [Q&A] [answered]
Assuming I have some native C++ code that will be compiled as a library and shipped to customers. In that library I'd like to call .NET code provided by the user. Assuming the user is using .NET >= 6.0 and the C++ code uses hostfxr, will my library be able to load .NET assemblies generated by newer versions? And if so, is that guaranteed somewhere? I'm thinking of something similar to https://docs.python.org/3/c-api/stable.html#stable-application-binary-interface

**Comment by vitek-karas** [comment, marked as answer]
The API exposed by `hostfxr` are considered stable, even across major versions. You can find the documentation how to use them here: https://learn.microsoft.com/en-us/dotnet/core/tutorials/netcore-hosting.

**Reply by mvtec-bergdolll** [reply]
Thanks for the quick reply, is this property documented anywhere?

**Reply by vitek-karas** [reply]
I honestly don't know where we document intended stability of our APIs. As far as I know pretty much all APIs are considered stable, unless they're marked as preview/experimental/...

Maybe @agocke or @terrajobst would know?

**Reply by jkotas** [reply]
https://learn.microsoft.com/en-us/dotnet/core/versions/#net-runtime-compatibility is our general statement about compatibility. It links to more detailed docs that talk about API breaking changes.

**Reply by mvtec-bergdolll** [reply]
@jkotas @vitek-karas  just to clarify. The linked documentation, as far as I can tell talks about **API** and not **ABI** compatibility. A stable API does not automatically imply a stable ABI.

For the outlined use-case of using hostfxr, a stable ABI is required, not only a stable API.
